template/{% if has_ai_rules %}AGENTS.md{% endif %}.jinja

```diff
@@ -0,0 +1,836 @@
+# JupyterLab Extension Development
+
+This guide provides coding standards and best practices for developing JupyterLab extensions. Follow these rules to align with community standards and keep your extension maintainable.
+
+**Extension type**: {{ kind }}
+
+## External Documentation
+
+For complete API reference and advanced topics, see:
+
+- [JupyterLab Extension Developer Guide](https://jupyterlab.readthedocs.io/en/stable/extension/extension_dev.html)
+- [JupyterLab Extension Examples](https://github.com/jupyterlab/extension-examples){% if kind == 'frontend-and-server' %}
```

Comment:
Wondering if it could also be useful to point to the JupyterLab repo, which implements many extensions and could be a good reference for common patterns?